### ISSUE TITLE
OsNet with triplet loss

### ISSUE THREAD

**Issue by DYT-code** [KaiyangZhou/deep-person-reid#246]
Hello Kaiyang,
I am just curious about if it's possible to train the OsNet with the triplet loss since I have to do the reid work in an online fashion. I've checked most of the issues of "triplet loss" and most of the models are ResNet50. At the moment I use the reid network TriNet which is introduced in "In Defense of the Triplet Loss for Person Re-identification". Since the reported performance of OsNet is much beyond the network TriNet. I would like to try to replace the ResNet50 with your OsNet in my reid network. 
Do you have any experience in training the OsNet with triplet loss and how should the configuration look like? Is there any pretrained model already available?
Thank you.

**Comment by KaiyangZhou** [comment]
We only have the osnet model pretrained using softmax. Triplet loss alone couldn't compete with softmax.

If you have enough data, softmax training could potentially give you a strong enough backbone so you could simply freeze the model as a feature extractor.

For online learning (I guess you have new classes coming in from time to time?), I think it might be difficult to just fine-tune the network using new classses only as this could damage the pretrained layers which already learn some good features from old images.

One solution is to fine-tune the network (maybe only for the last fc layer in osnet) using old+new classes using softmax, or softmax+triplet. For softmax training, it'd be better to train the classification layer first then open all network layers for fine-tuning.

If you are doing reid for people, I would suggest using the latest OSNet-AIN model (https://arxiv.org/abs/1910.06827), which is very good for cross-domain scenarios (this model is also suitable for non-people reid). Use the model pretrained on MSMT17 (all images). It's available in the model zoo.
